classes/class-post-vc-pub.php

```diff
@@ -0,0 +1,321 @@
+			return;
+		} else if ( ! isset( $post->post_status ) || 'publish' != $post->post_status ) {
+			return;
+		} else if ( wp_is_post_autosave( $post_id ) || wp_is_post_revision( $post_id ) ) {
+			return;
+		} else if ( ! $this->should_pub_vc( $post->ID ) ) {
+			return;
+		}
+
+		$is_valid_nonce = isset( $_POST['civil_pub_vc_nonce'] ) && wp_verify_nonce( $_POST['civil_pub_vc_nonce'], 'civil_pub_vc_action' );
+		if ( ! $is_valid_nonce || ! isset( $_POST[ PUB_VC_POST_FLAG ] ) ) {
+			return;
+		}
+
+		if ( ! get_post_meta( $post_id, POST_UUID_META_KEY, true ) ) {
+			add_post_meta( $post_id, POST_UUID_META_KEY, generate_uuid_v4() );
+		}
+
+		$vc_data = $this->generate_vc_metadata( $post );
+
+		$vc_log = get_option( VC_LOG_OPTION_KEY, '' );
+		$vc_log .= "generating VC for post $post->ID\ndata: " . json_encode( $vc_data, JSON_UNESCAPED_SLASHES );
+
+		try {
+			$jwt = $this->remote_sign_vc( $vc_data );
+			$vc_log .= "\nJWT: $jwt\n\n";
+			update_post_meta( $post_id, LAST_VC_PUB_DATE_META_KEY, $post->post_modified_gmt );
+		} catch ( \Exception $e ) {
+			$vc_log .= "\nFAILED: " . $e->getMessage() . "\n\n";
+		}
+
+		update_option( VC_LOG_OPTION_KEY, $vc_log );
+
+		// @TODO/tobek Now actually publish the VC JWT somewhere
+	}
+
+	/**
+	 * Generate post metadata to go inside VC
+	 *
+	 * @param \WP_Post|int $post The post object.
+	 * @return array Associative array of metadata.
+	 */
+	public function generate_vc_metadata( $post ) {
+		$post = get_post( $post );
+
+		$primary_category    = '';
+		$primary_category_id = get_post_meta( $post->ID, 'primary_category_id', true );
+
+		if ( ! empty( $primary_category_id ) ) {
+			$primary_category_term = get_term_by( 'id', $primary_category_id, 'category' );
+
+			if ( $primary_category_term instanceof \WP_Term ) {
+				$primary_category = $primary_category_term->slug;
+			}
+		}
+
+		$revision_content_hash = $this->hash( $post->post_content );
+
+		$post_uuid = get_post_meta( $post->ID, POST_UUID_META_KEY, true );
+		$site_url = site_url();
+		if ( false === strpos( $site_url, '?' ) ) {
+			$uuid_url = "$site_url?" . UUID_PERMALINK_QUERY . "=$post_uuid";
+		} else {
+			$uuid_url = "$site_url&" . UUID_PERMALINK_QUERY . "=$post_uuid";
+		}
+
+		return array(
+			'publishedContent' => array(
+				'title'               => $post->post_title,
+				'canonicalUrl'        => get_permalink( $parent_post->ID ),
+				'slug'                => $post->post_name,
+				'description'         => $post->post_excerpt,
+				'contributors'        => $this->get_contributor_data( $post ),
+				'tags'                => wp_get_post_tags( $post->ID, array( 'fields' => 'slugs' ) ),
+				'primaryTag'          => $primary_category,
+				'revisionDate'        => $post->post_modified_gmt,
+				'originalPublishDate' => $post->post_date_gmt,
+				'revisionContentHash' => $revision_content_hash,
+				'uuidUrl'             => $uuid_url,
+				'postUuid'            => $post_uuid,
+				'revisionUuid'        => generate_uuid_v4(),
```

Comment:
@tobek It's not a problem right now, but as we decided to adopt the attribute names as close as possible to schema.org, I suggest to review the properties names here as well. I've been doing some changes already in my code and you can take a look so we can use the same terminology. One example [here](https://github.com/uport-project/vc-explorer-content-service/blob/52c84517846632ce790e930466432a32fc92b451/src/graphql/Query.ts#L31).

Reply:
Updated as per our convo about schema, should all match up now and added stuff that was missing like `publisher` and `rawContentURL`.